app/controllers/users_controller.rb

```diff
@@ -35,67 +36,55 @@ def profile
 
   def show
     @user = User.find(params[:id])
-    @authorized_user = authorized_user(@user)
-    @user_invitable_groups = invitable_groups
-    @group_invitations = !@user_invitable_groups.empty? && !@authorized_user
-    # Groups owned by the current user && it is not the current user's profile
-    if @group_invitations
-      @invitation = Invitation.new
-    end
+    @invitable_groups = @user.invitable_groups(current_user)
+    @invitation = Invitation.new
   end
 
   def edit
   end
 
   def update
-    segment_string = params[:commit].split(' ')[1]
-    current_pass = params[:current_password] || params[:user][:password]
+    segment_string = params[:commit].split(' ').last
+    current_pass = params[:user][:current_password] || params[:current_password]
+    @user.skip_pass_strength = true unless segment_string == 'Password'
     if authenticate_user(current_pass) && @user.update(validate_params(segment_string))
       flash[:notice] = "#{segment_string} successfully updated."
       redirect_to profile_path
     else
-      flash[:warning] = "An error occurred, please try again."
+      flash[:warning] = @user.errors.full_messages.to_sentence
       render 'edit'
     end
   end
 
   private
 
   def authenticate_user(current_pass)
-    !!@user.authenticate(current_pass)
+    if @user.authenticate(current_pass)
```

Comment:
I'm not convinced this method belongs in the controller. Should the controller need to know how to authenticate a user or just if the user has been authenticated? Let's discuss further later. (I don't think this is a high-priority change)

Reply:
I will create an issue for this.

Reply:
#24